src/native/fs/fs_napi.cpp

```diff
@@ -11,12 +11,69 @@
 #include <vector>
 #include <math.h>
 #include <unistd.h>
+#include <map>
+#include <thread>
 
 namespace noobaa
 {
 
 DBG_INIT(0);
 
+const static std::map<std::string,int> flags_to_case = {
+    {"r",0},
+    {"rs",1},
+    {"sr",2}, 
+    {"r+",3},
+    {"rs+",4},
+    {"sr+",5},
+    {"w",6},
+    {"wx",7},
+    {"xw",8},
+    {"w+",9},
+    {"wx+",10},
+    {"xw+",11},
+    {"a",12},
+    {"ax",13},
+    {"xa",14},
+    {"as",15},
+    {"sa",16},
+    {"a+",17},
+    {"ax+",18},
+    {"xa+",19},
+    {"as+",20},
+    {"sa+",21}
+};
```

Comment:
Why not map directly to the flags value? why do we need this intermediate index values:
```
{"r", O_RDONLY},
{"rs", O_RDONLY | O_SYNC},
{"sr", O_RDONLY | O_SYNC},
{"r+", O_RDWR},
...
```